include/os/freebsd/spl/sys/condvar.h

```diff
@@ -157,14 +161,21 @@ cv_timedwait_sig_hires(kcondvar_t *cvp, kmutex_t *mp, hrtime_t tim,
 		tim += hrtime;
 
 	if (hrtime >= tim)
-		return (tim - hrtime);
+		return (-1);
 
 	sbt = zfs_nstosbt(tim);
 	rc = cv_timedwait_sig_sbt(cvp, mp, sbt, zfs_nstosbt(res), C_ABSOLUTE);
 
-	KASSERT(rc == EWOULDBLOCK || rc == EINTR || rc == ERESTART ||
-	    rc == 0, ("unexpected rc value %d", rc));
-	return (tim - gethrtime());
+	switch (rc) {
+	case EWOULDBLOCK:
+		return (-1);
+	case EINTR:
+	case ERESTART:
+		return (0);
+	default:
+		KASSERT(rc == 0, ("unexpected rc value %d", rc));
+		return (MAX(1, tim - gethrtime()));
```

Comment:
Looking on illumos code, they return -1 if time delta <= 0.  Though it may depend on really desired semantics in case if timeout and signal happened about same time.

Reply:
I've matched that behavior.